### PR TITLE
Add different styles

### PR DESCRIPTION
**Pull request by zadr**
Add support for changing the style in the app, where style is defined by a unified set of fonts + colors picked from a central location.

PR is kinda big, but, basically all the work is in one of two categories:
1. plumbing to get the `Style` object available everywhere it needed to be
2. removing `UIFont` and `UIColor` calls in favor of asking the style for a value to use

Plumbing works by having the root `SwipableNavigationContainer` pass a `Style` to all view controllers that conform to the `Styleable` protocol (with subclasses added for `UIPageViewController` and `UINavigationController` to help out in passing data along). View controllers than store this or pass it on to internal views as needed.

The root view also listens for `NSUserDefaultsDidChangeNotification`s to know when to pass a new style to the view controllers again for live updates of things.

All colors + fonts are defined on a `Style` protocol which is implemented by three concrete subclasses at the moment:  Affogato (classic look), Espresso (a nice dark brown look) and Cherry Mocha (dark mode).

I'm not attached to any of these looks and we should change 'em around as needed to look dope.

Screenshots:

**Affogato**
Simulator Screen Shot - iPhone Xʀ - 2019-05-19 at 10 47 28
Simulator Screen Shot - iPhone Xʀ - 2019-05-19 at 10 47 31

**Espresso**
Simulator Screen Shot - iPhone Xʀ - 2019-05-19 at 10 47 39
Simulator Screen Shot - iPhone Xʀ - 2019-05-19 at 10 47 42

**Cherry Mocha**
Simulator Screen Shot - iPhone Xʀ - 2019-05-19 at 10 47 49
Simulator Screen Shot - iPhone Xʀ - 2019-05-19 at 10 47 52

### CODE REVIEW

**Comment by rodericj** [comment]
Oh noes. It conflicts. Maybe with keita’s latest.

**Comment by zadr** [comment]
all fixed

**Comment by rodericj** [comment]
I’ve not had a chance to look. But in the images have the fonts changed?

**Comment by zadr** [comment]
yup, all unified in the style now.

everything support dynamic type now as well, which makes for a nice side effect

File: SF iOS/SF iOS/AppDelegate.m [hunk 1]
```diff
@@ -81,8 +81,7 @@ - (void)application:(UIApplication *)application performFetchWithCompletionHandl
 }
 
 //MARK: - Configure cache
-- (void)setSharedNetworkCacheMemoryMegabytes:(NSInteger)memoryMiB diskMegabytes:(NSInteger)diskMiB
-{
+- (void)setSharedNetworkCacheMemoryMegabytes:(NSInteger)memoryMiB diskMegabytes:(NSInteger)diskMiB {
```
Comment (rodericj) [inline SF iOS/SF iOS/AppDelegate.m:84]: very nice

**Comment by rodericj** [comment]
We lost some shadows in the event feed unfortunately. Can we 🌮 bout that?

**Comment by rodericj** [comment]
Screen Shot 2019-05-21 at 10 51 26 PM
Maybe this is just a style that's missing an attribute or something. I find it a bit hard to read.

File: SF iOS/SF iOS/AppDelegate.m (hunk 1 above)
Comment (mcritz) [reply SF iOS/SF iOS/AppDelegate.m:84]: Broken. Put it back. /s

**Comment by mcritz** [comment]
> We lost some shadows in the event feed unfortunately. Can we 🌮 bout that?

Must-fix

File: SF iOS/SF iOS/AppDelegate.m (hunk 1 above)
Comment (rodericj) [reply SF iOS/SF iOS/AppDelegate.m:84]: Works perfectly

**Review by rodericj — CHANGES_REQUESTED** [review]
Looks like some UI got lost.

**Comment by zadr** [comment]
> We lost some shadows in the event feed unfortunately. Can we 🌮 bout that?

fixed

**Comment by zadr** [comment]
also changed the tint for the icons, lmk if anything else needs to be done here

**Comment by rodericj** [comment]
Also yes this does in fact need to be rebased again :/

**Comment by rodericj** [comment]
Let's do it. I made changes over in #127

**Review by rodericj — APPROVED** [review]

**Pull request merged**